vectors/generate_test_vectors.py

```diff
@@ -0,0 +1,272 @@
+import json
+import os
+
+from umbral import pre
+from umbral.keys import UmbralPrivateKey
+from umbral.signing import Signer
+from umbral.curvebn import CurveBN
+from umbral.point import Point, unsafe_hash_to_point
+from umbral.config import set_default_curve, default_params
+
+
+#######################
+# Auxiliary functions #
+#######################
+
+def hexlify(data):
+    if isinstance(data, int):
+        return hex(data)[2:]
+    try:
+        return data.to_bytes().hex()
+    except AttributeError:
+        return bytes(data).hex()
+
+def create_test_vector_file(vector, filename, generate_again=False):
+    path = os.path.join(os.path.dirname(__file__), filename)
+
+    mode = 'w' if generate_again else 'x'
+    try:
+        with open(path, mode) as f:
+            json.dump(vector, f, indent=2)
+    except FileExistsError:
```

Comment:
Hmm - when is this getting raised?

Reply:
The whole idea of this method is the following: Since the generation of test vectors is always randomized, new test vector files would be created every time the script runs, overwriting the old ones. I don't want that, since it could lead to the test not detecting a change between versions (e.g., changing the public parameters). So, we use the mode 'w' when we want to overwrite the file no matter what, and the mode 'x' when we only want to write the file if it doesn't exist. The latter raises FileExistsError if the file exist, which for us simply means that there's nothing to do here.